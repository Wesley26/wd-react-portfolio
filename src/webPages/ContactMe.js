import React, { useState } from 'react'
import emailjs from 'emailjs-com';
import ReCAPTCHA from "react-google-recaptcha";
import { commonContactFormText, contactMeHeaderText, contactMeInstructionsA, contactMeInstructionsB, 
        contactMeInstructionsC, contactMeInstructionsD, contactMeInstructionsE, contactMeInstructionsF,
        contactMeInstructionsG, contactMeInstructionsH,
        SERVICE_ID, TEMPLATE_ID, USER_ID,
        RECAPTCHA_CLIENT_SIDE_KEY, } from './contentMaster/PageContents.js';

function ContactMe() {

    /**
     * botCheck, setBotCheck hooks as a string to match the TailwindCSS className
     * to control making the type send form input JSX element be "visible" or "hidden"
     */
    const [botCheck, setBotCheck] = useState("hidden bg-transparent");

    /**
     * formInput, setFormInput - Hook recieves user input for input name: 
     * subject, name, email, and message
     */
    const [formInput, setFormInput] = useState({subject: '', name: '', email: '', message: ''});
    let allowSend = true; //boolean to check if user input is filled out legitamately

    const handleInputChange = e => setFormInput({
        /**
        *  On event (e), looks for the current target input JSX element on entire form,
        *  pushes the current target value into formInput as it spreads over entire 
        *  JSX input current target element
        */
        ...formInput, 
        [e.currentTarget.name]: e.currentTarget.value
    });

    const submitValue = () => {
        const formDetails = {
            "The Subject" : formInput.subject,
            "The Name" : formInput.name,
            "The Email" : formInput.email,
            "The Message" : formInput.message,
        };
        //console.dir(formDetails); //Uncomment to see the object contents

        if (!formDetails) {
            alert("An error has occurred with the form, please email me for further assistance.");
            allowSend = false;
            return; //check to ensure the formDetails object exists, if not - do not send email
        };
        
        if (!formInput.subject ||
            !formInput.name ||
            !formInput.email ||
            !formInput.message ) {
            alert("ALL fields must be filled in to proceed.");
            allowSend = false;
            return; //check to ensure all fields are filled in, if not - do not send email
        };
    };

    let sendEmail = e => {
        e.preventDefault();

        if (allowSend === true) {
            emailjs.sendForm(SERVICE_ID, TEMPLATE_ID, e.target, USER_ID) //sends the form

            .then((result) => {
                console.log(result.text);
                alert("Your email has been successfully sent! You will hear back from me very soon.");
            }, (error) => {
                console.log(error.text);
                alert("An error has occurred with the form, please email me for further assistance.");
            });

            e.target.reset(); //reset the form
        };
    };

    const onPass = () => {
        //Passed reCaptcha - legitamate pass
        //add tailwindCSS class 'visible'
        setBotCheck("visible bg-transparent");
    };

    const onFail = () => {
        //Failed reCaptcha - time-out or any fail reason
        //add tailwindCSS class 'hidden'
        setBotCheck("hidden bg-transparent");
    };
  
    return (
      <div className="bg-gray-100 m-6 pt-6 pl-6 pr-6 pb-16 overflow-auto">
          <div className="md:ml-32 md:mr-32 lg:ml-64 lg:mr-64 p-3 grid grid-cols-1 items-center text-center">
              <h3 className="p-3 font-bold bg-transparent">
                {contactMeHeaderText}
              </h3>
              <p className="p-3">
                {contactMeInstructionsA}
              </p>
              <div className="pl-3 pr-3 pt-3 md:flex md:justify-start text-left">
                    <p className="p-3">
                        <b>{commonContactFormText.subjectText}</b>{contactMeInstructionsB}
                    </p>
                    <p className="p-3">
                        <b>{commonContactFormText.nameText}</b>{contactMeInstructionsC}
                    </p>
                    <p className="p-3">
                        <b>{commonContactFormText.emailText}</b>{contactMeInstructionsD}
                    </p>
                    <p className="p-3">
                        <b>{commonContactFormText.messageText}</b>{contactMeInstructionsE}
                    </p>
              </div>
              <p className="pt-3">
                {contactMeInstructionsF}<b>{contactMeInstructionsG}</b>{contactMeInstructionsH}
              </p>
          </div>
          <br>
          </br>
          <div className="bg-header-blue p-6 flex justify-center place-items-center">
      
            <form
                className="bg-gray-300 grid grid-cols-1 items-center"
                onSubmit={sendEmail}
            >
                <div className="pl-64 pr-64 pt-3 pb-3">
                    <br>
                    </br>
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="p-3"><b>{commonContactFormText.subjectText}</b></label>
                    <input
                        type="text" 
                        name ="subject"
                        size="25"
                        onChange={handleInputChange}
                    />
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="p-3"><b>{commonContactFormText.nameText}</b></label>
                    <input 
                        type="text" 
                        name="name"
                        size="25"
                        onChange={handleInputChange}
                    />
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="p-3"><b>{commonContactFormText.emailText}</b></label>
                    <input 
                        type="email" 
                        name="email"
                        size="25"
                        onChange={handleInputChange}
                    />
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="p-3 flex justify-center"><b>{commonContactFormText.messageText}</b></label>
                    <textarea
                        name="message"
                        rows="8"
                        cols="25"
                        onChange={handleInputChange}
                    />
                </div>

                <div className="p-3 flex justify-center">
                    <ReCAPTCHA
                        sitekey={RECAPTCHA_CLIENT_SIDE_KEY}
                        onChange={onPass}
                        onExpired={onFail}
                        onErrored={onFail} 
                    />
                </div>

                <div className="p-3 flex justify-center">
                    <p className="p-3 rounded-full hover:bg-gray-500">
                    <input
                        className={botCheck}
                        onClick={submitValue}
                        type="submit" 
                        value="Send"
                    />
                    </p>
                </div>
            </form>

          </div>
    </div>
    );
};

export default ContactMe;