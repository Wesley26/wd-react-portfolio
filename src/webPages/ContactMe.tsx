import { FC, useState } from 'react'
import emailjs from 'emailjs-com';
import ReCAPTCHA from "react-google-recaptcha";
import PageContents from './contentMaster/PageContents';

const ContactMe:FC = () => {

    const pageContents = PageContents();

    /**
     * botCheck, setBotCheck hooks as a string to match the TailwindCSS className
     * to control making the type send form input JSX element be "visible" or "hidden"
     */
    const [botCheck, setBotCheck] = useState<string>("hidden bg-transparent font-semibold");

    /**
     * formInput, setFormInput - Hook recieves user input for input name: 
     * subject, name, email, and message
     */
    const [formInput, setFormInput] = 
    useState<{subject: string, name: string, email: string, message: string}>({subject: '', name: '', email: '', message: ''});
    let allowSend = true; //boolean to check if user input is filled out legitamately

    const handleInputChange = (e: any) => setFormInput({
        /**
        *  On event (e), looks for the current target input JSX element on entire form,
        *  pushes the current target value into formInput as it spreads over entire 
        *  JSX input current target element
        */
        ...formInput, 
        [e.currentTarget.name]: e.currentTarget.value
    });

    const submitValue = () => {
        const formDetails = {
            "The Subject" : formInput.subject,
            "The Name" : formInput.name,
            "The Email" : formInput.email,
            "The Message" : formInput.message,
        };
        //console.dir(formDetails); //Uncomment to see the object contents

        if (!formDetails) {
            alert("An error has occurred with the form, please email me for further assistance.");
            allowSend = false;
            return; //check to ensure the formDetails object exists, if not - do not send email
        };
        
        if (!formInput.subject ||
            !formInput.name ||
            !formInput.email ||
            !formInput.message ) {
            alert("ALL fields must be filled in to proceed.");
            allowSend = false;
            return; //check to ensure all fields are filled in, if not - do not send email
        };
    };

    let sendEmail = (e: any) => {
        e.preventDefault();

        const SERVICE_ID = pageContents.apiContent.SERVICE_ID as string;
        const TEMPLATE_ID = pageContents.apiContent.TEMPLATE_ID as string;
        const USER_ID = pageContents.apiContent.USER_ID as string;

        if (allowSend === true) {
            emailjs.sendForm(SERVICE_ID, TEMPLATE_ID, e.target, USER_ID) //sends the form

            .then((result) => {
                console.log(result.text);
                alert("Your email has been successfully sent! You will hear back from me very soon.");
            }, (error) => {
                console.log(error.text);
                alert("An error has occurred with the form, please email me for further assistance.");
            });

            e.target.reset(); //reset the form
        };
    };

    const onPass = () => {
        //Passed reCaptcha - legitamate pass
        //add tailwindCSS class 'visible'
        setBotCheck("visible bg-transparent font-semibold");
    };

    const onFail = () => {
        //Failed reCaptcha - time-out or any fail reason
        //add tailwindCSS class 'hidden'
        setBotCheck("hidden bg-transparent font-semibold");
    };
  
    return (
      <div className="font-body bg-body-lightGray m-6 pt-6 pl-6 pr-6 pb-16 overflow-auto shadow-xl">
          <div className="md:ml-12 md:mr-12 lg:ml-40 lg:mr-40 p-3 grid grid-cols-1 items-center text-center">
              <h3 className="font-title text-xl p-3 font-bold bg-transparent">
                {pageContents.contactMeHeaderText}
              </h3>
              <p className="text-lg p-3">
                {pageContents.contactMeInstructionsA}
              </p>
              <div className="text-lg pl-3 pr-3 pt-3 md:flex md:justify-start text-left">
                    <p className="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300">
                        <b className="font-title text-xl">{pageContents.commonContactFormText.subjectText}</b>{pageContents.contactMeInstructionsB}
                    </p>
                    <p className="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300">
                        <b className="font-title text-xl">{pageContents.commonContactFormText.nameText}</b>{pageContents.contactMeInstructionsC}
                    </p>
                    <p className="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300">
                        <b className="font-title text-xl">{pageContents.commonContactFormText.emailText}</b>{pageContents.contactMeInstructionsD}
                    </p>
                    <p className="p-3 border-b-4 md:border-b-0 border-gray-300">
                        <b className="font-title text-xl">{pageContents.commonContactFormText.messageText}</b>{pageContents.contactMeInstructionsE}
                    </p>
              </div>
              <p className="text-lg pt-3">
                {pageContents.contactMeInstructionsF}<b>{pageContents.contactMeInstructionsG}</b>{pageContents.contactMeInstructionsH}
              </p>
          </div>
          <br>
          </br>
          <div className="bg-body-gray p-6 flex justify-center place-items-center shadow-lg">
      
            <form
                className="bg-body-form md:pl-16 md:pr-16 lg:pl-64 lg:pr-64 grid grid-cols-1 items-center overflow-auto shadow-md"
                onSubmit={sendEmail}
            >
                <div className="sm:pl-48 sm:pr-48 md:pl-64 md:pr-64 pt-3 pb-3">
                    <br>
                    </br>
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="font-title text-xl p-3"><b>{pageContents.commonContactFormText.subjectText}</b></label>
                    <input
                        className="shadow-lg"
                        type="text" 
                        name ="subject"
                        size={26}
                        placeholder={pageContents.placeholderText.placeSubject}
                        onChange={handleInputChange}
                    />
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="font-title text-xl p-3"><b>{pageContents.commonContactFormText.nameText}</b></label>
                    <input
                        className="shadow-lg"
                        type="text" 
                        name="name"
                        size={26}
                        placeholder={pageContents.placeholderText.placeName}
                        onChange={handleInputChange}
                    />
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="font-title text-xl p-3"><b>{pageContents.commonContactFormText.emailText}</b></label>
                    <input
                        className="shadow-lg"
                        type="email" 
                        name="email"
                        size={26}
                        placeholder={pageContents.placeholderText.placeEmail}
                        onChange={handleInputChange}
                    />
                </div>
                <div className="p-3 grid justify-items-center">
                    <label className="font-title text-xl p-3 flex justify-center"><b>{pageContents.commonContactFormText.messageText}</b></label>
                    <textarea
                        className="shadow-lg"
                        name="message"
                        rows={8}
                        cols={26}
                        placeholder={pageContents.placeholderText.placeMessage}
                        onChange={handleInputChange}
                    />
                </div>

                <div className="p-3 flex justify-center">
                    <ReCAPTCHA
                        sitekey={pageContents.apiContent.RECAPTCHA_CLIENT_SIDE_KEY}
                        onChange={onPass}
                        onExpired={onFail}
                        onErrored={onFail} 
                    />
                </div>

                <div className="p-3 flex justify-center">
                    <div className="font-title text-xl p-3 border-solid border-l-4 border-r-4 border-gray-300 rounded-full hover:bg-gray-400 active:bg-gray-500">
                        <input
                            className={botCheck}
                            onClick={submitValue}
                            type="submit" 
                            value="Send Information"
                        />
                    </div>
                </div>
            </form>

          </div>
    </div>
    );
};

export default ContactMe;