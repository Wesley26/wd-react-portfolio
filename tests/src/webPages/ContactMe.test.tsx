import { render } from '@testing-library/react';
import ContactMe from '../../../src/webPages/ContactMe';
import '../../src/index.css';

it(`render ContactMe.tsx test component`, () => {

    const view = (render(<ContactMe />));

    expect(view).toMatchSnapshot();

});