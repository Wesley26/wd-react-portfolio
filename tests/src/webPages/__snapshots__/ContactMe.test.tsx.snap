// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ContactMe.tsx test component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="font_body bg-body_lightGray m-6 pt-6 pl-6 pr-6 pb-16 overflow-auto shadow-xl"
      >
         
        <div
          style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
        >
          <div
            class="md:ml-12 md:mr-12 lg:ml-40 lg:mr-40 p-3 grid grid-cols-1 items-center text-center"
          >
            <h3
              class="font-title text-xl p-3 font-bold bg-transparent"
            >
              Contact Me
            </h3>
            <p
              class="text-lg p-3"
            >
              Thank you for your interest in reaching out to me! Complete
                                    this form to send an email and I will reply.
                                    For my reply to be sent back to you, be sure to include all the
                                    following details on this form:
            </p>
            <div
              class="text-lg pl-3 pr-3 pt-3 md:flex md:justify-start text-left"
            >
              <p
                class="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300"
              >
                <b
                  class="font-title text-xl"
                >
                  Subject
                </b>
                 - This is the subject line of the email I receive.
              </p>
              <p
                class="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300"
              >
                <b
                  class="font-title text-xl"
                >
                  Name
                </b>
                 - You enter your full name on this line. You may use your first, last, or full name.
              </p>
              <p
                class="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300"
              >
                <b
                  class="font-title text-xl"
                >
                  Email
                </b>
                 - You enter the email you prefer me to reply back to on this line (example: your_email_here@mail.com).
              </p>
              <p
                class="p-3 border-b-4 md:border-b-0 border-gray-300"
              >
                <b
                  class="font-title text-xl"
                >
                  Message
                </b>
                 - You enter in your message to me in this text box.
              </p>
            </div>
            <p
              class="text-lg pt-3"
            >
              Be sure to fill out 
              <b>
                ALL
              </b>
               provided fields below. If the ReCaptcha test
                                    does not appear, refresh this page. You need to verify that you are human
                                    to use this email form. A send button will only appear on the bottom of 
                                    this form if you complete verification.
            </p>
          </div>
          <br />
          <div
            class="bg-body_gray p-6 flex justify-center place-items-center shadow-lg"
          >
            <form
              class="bg-body_form md:pl-16 md:pr-16 lg:pl-64 lg:pr-64 grid grid-cols-1 items-center overflow-auto shadow-md"
            >
              <div
                class="sm:pl-48 sm:pr-48 md:pl-64 md:pr-64 pt-3 pb-3"
              >
                <br />
              </div>
              <div
                class="p-3 grid justify-items-center"
              >
                <label
                  class="font-title text-xl p-3"
                >
                  <b>
                    Subject
                  </b>
                </label>
                <input
                  class="shadow-lg"
                  name="subject"
                  placeholder="Add A Subject"
                  size="26"
                  type="text"
                />
              </div>
              <div
                class="p-3 grid justify-items-center"
              >
                <label
                  class="font-title text-xl p-3"
                >
                  <b>
                    Name
                  </b>
                </label>
                <input
                  class="shadow-lg"
                  name="name"
                  placeholder="Add A Name"
                  size="26"
                  type="text"
                />
              </div>
              <div
                class="p-3 grid justify-items-center"
              >
                <label
                  class="font-title text-xl p-3"
                >
                  <b>
                    Email
                  </b>
                </label>
                <input
                  class="shadow-lg"
                  name="email"
                  placeholder="email@mail.com"
                  size="26"
                  type="email"
                />
              </div>
              <div
                class="p-3 grid justify-items-center"
              >
                <label
                  class="fon-title text-xl p-3 flex justify-center"
                >
                  <b>
                    Message
                  </b>
                </label>
                <textarea
                  class="shadow-lg"
                  cols="26"
                  name="message"
                  placeholder="Hello World!"
                  rows="8"
                />
              </div>
              <div
                class="p-3 flex justify-center"
              >
                <div />
              </div>
              <div
                class="p-3 flex justify-center"
              >
                <div
                  class="font-title text-xl p-3 border-solid border-l-4 border-r-4 border-gray-300 rounded-full hover:bg-gray-400 active:bg-gray-500"
                >
                  <input
                    class="hidden bg-transparent font-semibold"
                    type="submit"
                    value="Send Information"
                  />
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <script
      src="https://www.google.com/recaptcha/api.js?onload=onloadcallback&render=explicit"
    />
  </body>,
  "container": <div>
    <div
      class="font_body bg-body_lightGray m-6 pt-6 pl-6 pr-6 pb-16 overflow-auto shadow-xl"
    >
       
      <div
        style="opacity: 0; transform: translateX(0px) translateY(-20px) translateZ(0);"
      >
        <div
          class="md:ml-12 md:mr-12 lg:ml-40 lg:mr-40 p-3 grid grid-cols-1 items-center text-center"
        >
          <h3
            class="font-title text-xl p-3 font-bold bg-transparent"
          >
            Contact Me
          </h3>
          <p
            class="text-lg p-3"
          >
            Thank you for your interest in reaching out to me! Complete
                                    this form to send an email and I will reply.
                                    For my reply to be sent back to you, be sure to include all the
                                    following details on this form:
          </p>
          <div
            class="text-lg pl-3 pr-3 pt-3 md:flex md:justify-start text-left"
          >
            <p
              class="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300"
            >
              <b
                class="font-title text-xl"
              >
                Subject
              </b>
               - This is the subject line of the email I receive.
            </p>
            <p
              class="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300"
            >
              <b
                class="font-title text-xl"
              >
                Name
              </b>
               - You enter your full name on this line. You may use your first, last, or full name.
            </p>
            <p
              class="p-3 border-b-4 md:border-b-0 md:border-r-4 border-gray-300"
            >
              <b
                class="font-title text-xl"
              >
                Email
              </b>
               - You enter the email you prefer me to reply back to on this line (example: your_email_here@mail.com).
            </p>
            <p
              class="p-3 border-b-4 md:border-b-0 border-gray-300"
            >
              <b
                class="font-title text-xl"
              >
                Message
              </b>
               - You enter in your message to me in this text box.
            </p>
          </div>
          <p
            class="text-lg pt-3"
          >
            Be sure to fill out 
            <b>
              ALL
            </b>
             provided fields below. If the ReCaptcha test
                                    does not appear, refresh this page. You need to verify that you are human
                                    to use this email form. A send button will only appear on the bottom of 
                                    this form if you complete verification.
          </p>
        </div>
        <br />
        <div
          class="bg-body_gray p-6 flex justify-center place-items-center shadow-lg"
        >
          <form
            class="bg-body_form md:pl-16 md:pr-16 lg:pl-64 lg:pr-64 grid grid-cols-1 items-center overflow-auto shadow-md"
          >
            <div
              class="sm:pl-48 sm:pr-48 md:pl-64 md:pr-64 pt-3 pb-3"
            >
              <br />
            </div>
            <div
              class="p-3 grid justify-items-center"
            >
              <label
                class="font-title text-xl p-3"
              >
                <b>
                  Subject
                </b>
              </label>
              <input
                class="shadow-lg"
                name="subject"
                placeholder="Add A Subject"
                size="26"
                type="text"
              />
            </div>
            <div
              class="p-3 grid justify-items-center"
            >
              <label
                class="font-title text-xl p-3"
              >
                <b>
                  Name
                </b>
              </label>
              <input
                class="shadow-lg"
                name="name"
                placeholder="Add A Name"
                size="26"
                type="text"
              />
            </div>
            <div
              class="p-3 grid justify-items-center"
            >
              <label
                class="font-title text-xl p-3"
              >
                <b>
                  Email
                </b>
              </label>
              <input
                class="shadow-lg"
                name="email"
                placeholder="email@mail.com"
                size="26"
                type="email"
              />
            </div>
            <div
              class="p-3 grid justify-items-center"
            >
              <label
                class="fon-title text-xl p-3 flex justify-center"
              >
                <b>
                  Message
                </b>
              </label>
              <textarea
                class="shadow-lg"
                cols="26"
                name="message"
                placeholder="Hello World!"
                rows="8"
              />
            </div>
            <div
              class="p-3 flex justify-center"
            >
              <div />
            </div>
            <div
              class="p-3 flex justify-center"
            >
              <div
                class="font-title text-xl p-3 border-solid border-l-4 border-r-4 border-gray-300 rounded-full hover:bg-gray-400 active:bg-gray-500"
              >
                <input
                  class="hidden bg-transparent font-semibold"
                  type="submit"
                  value="Send Information"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
